wiki/Beatmapping/Flow/en.md

```diff
@@ -0,0 +1,39 @@
+---
+tags:
+  - linear flow
+  - oval flow
+  - circular flow
+---
+
+# Flow
+
+The **flow** is a mapping concept that describes natural and intuitive movement of a player throughout the map. The flow, or lack thereof, is almost always related to the [osu!](/wiki/Game_mode/osu!) game mode beatmaps due to the two-dimensional playfield. Additionally, the osu! beatmaps offer a vast variety of [patterns](/wiki/Beatmap/Pattern) and slider shapes, which may be used by mappers to direct a player's cursor wherever they want to.
+
+If a mapper deliberately ignores the concepts of flow, or uses their opposites, they utilise [antiflow](/wiki/Beatmapping/Antiflow).
+
+## Key components
+
+While flow is mostly subjective, as movement preferences differ from one person to another, there are a few ideas that give the feeling and impression of flow on a majority of patterns or whole beatmaps.
+
+### Smooth movement
+
+The cursor movement required to pass the map is stable most of the time and doesn't contain sharp, abrupt turns, or fast change of pace. Often, a player's cursor follows a smooth curve that spans for several patterns. The curve contains lots of circular or semicircular segments (so-called *oval or circular flow*) going in different directions, mixed with "neutral", non-demanding patterns. Such patterns don't contribute to the general direction, or only change it slightly to ease the transition to the next pattern.
```

Comment:
might be worth mentioning that circular flow for too long in the same direction is viewed as uncomfortable because mousedrift